src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -78,13 +78,29 @@ sealed class ChannelState {
             is Syncing -> this@ChannelState.state
             else -> this@ChannelState
         }
+        maybeSignalSensitiveTask(oldState, newState)
         return when {
             // we only want to fire the PaymentSent event when we transition to Closing for the first time
             oldState is ChannelStateWithCommitments && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
             else -> emptyList()
         }
     }
 
+    /** Some transitions imply that we are in the middle of tasks that may require some time. */
+    private fun ChannelContext.maybeSignalSensitiveTask(oldState: ChannelState, newState: ChannelState) {
+        val spliceStatusBefore = (oldState as? Normal)?.spliceStatus
+        val spliceStatusAfter = (newState as? Normal)?.spliceStatus
+        when {
+            spliceStatusBefore !is SpliceStatus.InProgress && spliceStatusAfter is SpliceStatus.InProgress -> // splice initiated
+                staticParams.nodeParams._nodeEvents.tryEmit(SensitiveTaskEvents.TaskStarted(SensitiveTaskEvents.TaskIdentifier.InteractiveTx(spliceStatusAfter.spliceSession.fundingParams)))
+            spliceStatusBefore is SpliceStatus.InProgress && spliceStatusAfter !is SpliceStatus.WaitingForSigs -> // splice aborted before reaching signing phase
+                staticParams.nodeParams._nodeEvents.tryEmit(SensitiveTaskEvents.TaskEnded(SensitiveTaskEvents.TaskIdentifier.InteractiveTx(spliceStatusBefore.spliceSession.fundingParams)))
+            spliceStatusBefore is SpliceStatus.WaitingForSigs && spliceStatusAfter !is SpliceStatus.WaitingForSigs -> // splice leaving signing phase (successfully or not)
```

Comment:
The argument for considering the process done in the success case when leaving `SpliceStatus.WaitingForSigs` is that we have already either sent or received `tx_signatures` (not both). The splice could in theory be failed after restart, but the implementation on lightning-kmp and eclair guarantees that it will not be the case.

Reply:
That looks correct! Using the transitions on the `spliceStatus` really makes a lot of sense here :+1:

Reply:
> We have already either sent or received-then-sent tx_signatures (same for our peer). 

That even guarantees that we won't forget this splice, because either us or our peer has enough signatures to broadcast it. The only way we could forget it in that case is if it was double-spent, but 1) we don't detect that in lightning-kmp and 2) we explicitly avoid double-spending those in eclair